quickwit-cluster/src/cluster.rs

```diff
@@ -88,15 +104,19 @@ impl Cluster {
     /// Create a cluster given a host key and a listen address.
     /// When a cluster is created, the thread that monitors cluster events
     /// will be started at the same time.
-    pub fn new(host_key: Uuid, listen_addr: SocketAddr) -> anyhow::Result<Self> {
+    pub fn new(host_key: Uuid, listen_addr: SocketAddr) -> ClusterResult<Self> {
         info!( host_key=?host_key, listen_addr=?listen_addr, "Create new cluster.");
         let config = ArtilleryClusterConfig {
             cluster_key: CLUSTER_ID.as_bytes().to_vec(),
             listen_addr,
             ..Default::default()
         };
         let (artillery_cluster, _) =
-            ArtilleryCluster::new_cluster(host_key, config).map_err(|err| anyhow::anyhow!(err))?;
+            ArtilleryCluster::new_cluster(host_key, config).map_err(|err| {
+                ClusterError::CreateClusterError {
```

Comment:
We can be more accurate it...

err is ArtilleryError.
Can you match case on it, and if it is an IOError, we could return an error accordingly. 

Something 

```
enum ClusterError {
    #[error("Failed to bind UDP port :{port} for the gossip membership algorithm. Cause: {cause}.")]
    UDPPortBindingError {
        port: u16,
        #[source]
        cause: io::Error
  }
  ...

}
```

Reply:
How about this?

```
% cargo run -- serve --host=0.0.0.0 --index-uri=file:///home/minoru/tmp/quickwit_data/indices/wikipedia-idx --port=80
    Finished dev [unoptimized + debuginfo] target(s) in 0.21s
     Running `target/debug/quickwit serve --host=0.0.0.0 '--index-uri=file:///home/minoru/tmp/quickwit_data/indices/wikipedia-idx' --port=80`
Jul 16 15:32:24.034  INFO quickwit_cluster::cluster: Read existing host key. host_key=57aa5a51-56ef-4d76-9880-16e749788a8c host_key_path="/tmp/quickwit-host-key-0.0.0.0-80"
Jul 16 15:32:24.034  INFO quickwit_cluster::cluster: Create new cluster. host_key=57aa5a51-56ef-4d76-9880-16e749788a8c listen_addr=0.0.0.0:81
Command failed: Failed to bind UDP port :81 for the gossip membership algorithm. Cause: Permission denied (os error 13)

Caused by:
    Permission denied (os error 13)
```